src/NLog/Targets/FileTarget.cs

```diff
@@ -719,17 +721,26 @@ private void RefreshArchiveFilePatternToWatch()
                 else
                 {
                     this.fileAppenderCache.ArchiveFilePatternToWatch = null;
-
-                    if (this.appenderInvalidatorThread != null)
-                    {
-                        this.appenderInvalidatorThread.Abort();
-                        this.appenderInvalidatorThread = null;
-                    }
+                    this.StopAppenderInvalidatorThread();
                 }
             }
 #endif
         }
 
+        private void StopAppenderInvalidatorThread()
```

Comment:
I think some docs are nice, as this is a pretty difficult case (to test at least)